src/components/BlogList.jsx

```diff
@@ -1,22 +1,31 @@
 import BlogPost from "./BlogPost";
 import Pagination from "./Pagination";
-import React from "react";
+import React, { useState } from "react";
 import blogs from "../data/blogs.json";
 
 const PAGE_SIZES = [15, 25, 50, 100];
 
 function BlogList() {
-  const currentPaginationData = blogs.posts.slice(0, 15);
+  const [pageSize, setPageSize] = useState(PAGE_SIZES[0]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const firstBlog = (currentPage - 1) * pageSize;
+  const lastBlog = currentPage * pageSize;
+  const currentPaginationData = blogs.posts.slice(firstBlog, lastBlog);
```

Comment:
This value can be memoized/cached to avoid unnecessarily slicing the "blogs.posts" array every time the BlogList component re-renders. The useMemo() hook seems suitable.